include/boost/decimal/detail/apply_sign.hpp

```diff
@@ -34,7 +34,7 @@ template <typename Integer, typename Unsigned_Integer = detail::make_unsigned_t<
           std::enable_if_t<detail::is_signed_v<Integer>, bool> = true>
 constexpr auto make_positive_unsigned(Integer val) noexcept -> Unsigned_Integer
 {
-    return static_cast<Unsigned_Integer>(val < static_cast<Integer>(0) ? apply_sign(val) : val);
+    return static_cast<Unsigned_Integer>(val < static_cast<Integer>(static_cast<std::int8_t>(INT8_C(0))) ? apply_sign(val) : val);
```

Comment:
Is this necessary? We are already casting 0 to the target type.

Reply:
This is a subtle one. I believe since fundamental types such as `int`, `short` or `long` _might_ differ on `arm-none-eabi-gcc` than they are from other compilers, that the construction from `int128` is ambiguous. The compiler issues not a warning fut a hard error. I have not figured out the exact place to correct these. Probably a whole bunch of `enable_if`s on the ctors of the emulated 128-bit integer.

See image here.

![grafik](https://github.com/cppalliance/decimal/assets/2404721/accccefb-b6a2-45d8-9f2d-d79c502b9faa)

Reply:
We need to correct something or this will creep up on other compilers that have maybe even wider (nit lss-wide) types as well. So I need to stick with some kind of cast since these changes are not in the scope of the metal PR.

Reply:
That's annoying because all the integer constructors of int128 are fixed width types on purpose: https://github.com/cppalliance/decimal/blob/develop/include/boost/decimal/detail/emulated128.hpp#L600. I think you can drop the `INT8_C` since now it's being casted to int8_t twice.